File: api/src/main/java/dev/ulloasp/mlsuite/workspace/application/dto/WorkspacePermissionsDto.java
package dev.ulloasp.mlsuite.workspace.application.dto;

public record WorkspacePermissionsDto(
        boolean canViewWorkspace,
        boolean canViewOrganization,
        boolean canEditOrganization,
        boolean canDeleteOrganization,
        boolean canTransferOwnership,
        boolean canViewMembers,
        boolean canInviteMembers,
        boolean canManageMemberRoles,
        boolean canRemoveMembers,
        boolean canViewInvitations,
        boolean canManageInvitations,
        boolean canViewModels,
        boolean canCreateModels,
        boolean canEditModels,
        boolean canDeleteModels,
        boolean canRunPredictions,
        boolean canExportPredictions,
        boolean canReview,
        boolean canManageReviews,
        boolean canViewPlugins,
        boolean canManagePlugins) {
}

File: frontend/src/api/workspace/dtos/workspace-permissions-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

export interface WorkspacePermissionsDto {
  canViewWorkspace: boolean;
  canViewOrganization: boolean;
  canEditOrganization: boolean;
  canDeleteOrganization: boolean;
  canTransferOwnership: boolean;
  canViewMembers: boolean;
  canInviteMembers: boolean;
  canManageMemberRoles: boolean;
  canRemoveMembers: boolean;
  canViewInvitations: boolean;
  canManageInvitations: boolean;
  canViewTeams: boolean;
  canCreateTeams: boolean;
  canEditTeams: boolean;
  canDeleteTeams: boolean;
  canViewModels: boolean;
  canCreateModels: boolean;
  canEditModels: boolean;
  canDeleteModels: boolean;
  canRunPredictions: boolean;
  canExportPredictions: boolean;
  canManageReviewLinks: boolean;
  canViewPlugins: boolean;
  canManagePlugins: boolean;
}
